som/display.go

```diff
@@ -32,12 +32,24 @@ type svgElement struct {
 	Polygons []interface{}
 }
 
-// UMatrixSVG creates SVG representation of the U-Matrix of the given codebook.
-// codebook is a SOM codebook we're rendering the U-Matrix for,
-// dims are dimensions of the map grid, uShape is the shape of the grid unit,
-// title is the title of the output SVG, and writer is the io.Writter to write the SVG to.
-// UMatrixSVG returns error when the SVG could not be generated.
-func UMatrixSVG(codebook *mat64.Dense, dims []int, uShape string, title string, writer io.Writer) error {
+type textElement struct {
+	XMLName xml.Name `xml:"text"`
+	X       float64  `xml:"x,attr"`
+	Y       float64  `xml:"y,attr"`
+	Text    string   `xml:",innerxml"`
+}
+
+var colors = [][]int{{255, 0, 0}, {0, 255, 0}, {0, 0, 255}, {255, 255, 0}, {255, 0, 255}, {0, 255, 255}}
+
+// Creates an SVG representation of the U-Matrix of the given codebook.
+// codebook - the codebook we're displaying the U-Matrix for
+// dims - the dimensions of the grid
+// uShape - the shape of the grid
+// title - the title of the output SVG
+// writer - the io.Writter to write the output SVG to.
+// clusters - if the clusters are known (i.e. these are test data) they can be displayed providing the information in this map.
+// The map is: codebook vector row -> cluster number.  When clusters are not known (i.e. running with real data), just provide an empty map.
+func UMatrixSVG(codebook *mat64.Dense, dims []int, uShape string, title string, writer io.Writer, clusters map[int]int) error {
```

Comment:
You can group `uShape` and `title` in func header like this:

`uShape, title string`